blockstack/blockstack-browser
# Client-side transaction generation

This pull request enables client-side transaction generation using the newest version of Blockstack.js and the transaction broadcasting service. 

Testing steps:
1. Run `npm link` from blockstack.js repo on the `blockstack-operations` branch
2. Run `npm link blockstack` from the browser repo
3. Run `npm run mac` to build the new CORS proxy executable (which handles both http + https requests)
4. Run the native app from xcode
5. Enable development mode + Enable regtest mode
6. Pull the latest version of the integration test docker image using `docker pull quay.io/blockstack/integrationtests:develop`
7. Start the regtest docker container using the following command:
```
IMAGE=$(docker run -dt -p 6270:6270 -p 16269:16269 -p 18332:18332 -e BLOCKSTACK_TEST_CLIENT_RPC_PORT=6270 -e BLOCKSTACK_TEST_CLIENT_BIND=0.0.0.0 -e BLOCKSTACK_TEST_BITCOIND_ALLOWIP=0.0.0.0/0 quay.io/blockstack/integrationtests:develop blockstack-test-scenario --interactive 2 blockstack_integration_tests.scenarios.browser_env)
```
The docker container includes a transaction broadcasting service instance

8. Wait for docker container to fully start
9. Click on the Blockstack system tray icon to launch the browser in dev mode, this will now open the browser to `regtest` mode, where registrations transactions will be generated for the LOCAL_REGTEST env.
10. Create a new account
11. Go to `/wallet/send-core` and transfer out the bitcoin
12. Go to IDs -> Add username
13. Verify that the name is registered successfully.
